cloud/services/object_storage_buckets.go

```diff
@@ -11,7 +11,7 @@ import (
 	"github.com/linode/cluster-api-provider-linode/util"
 )
 
-func EnsureObjectStorageBucket(ctx context.Context, bScope *scope.ObjectStorageBucketScope) (*linodego.ObjectStorageBucket, error) {
+func EnsureAndUpdateObjectStorageBucket(ctx context.Context, bScope *scope.ObjectStorageBucketScope) (*linodego.ObjectStorageBucket, error) {
```

Comment:
I think there is an opportunity to simplify this nested if logic here, could we do a small refactor here so it is something like
```
if bucket == nil {
 return create bucket
}
if bucketAccess == bscope.Access { return bucket, nil }
update bucketAccess
return bucket,nil
```